### ISSUE TITLE
Title resolvers in Routing don't have access to other resolved data

### ISSUE THREAD

**Issue by Armenvardanyan95** [angular/angular#57752]
### Which @angular/* package(s) are the source of the bug?

router

### Is this a regression?

No

### Description

So, in recent versions, we are able to provide titles for routes via a string or a resolver function. This issue was discovered when trying to use resolved data as a resolver for the page title.

In general, dynamic titles often rely on some HTTP calls to produce meaningful titles, and the issue of using already resolved data comes up if we do not want to: 

1. repeat the same HTTP call twice
2. move the dynamic title setup logic into a component

In this example, we have a resolver that resolves a product item, and then a dynamic title resolver that wants to reuse that resolved data. Which, of course, makes sense, as I already pulled the product data and just want to pluck the title of the product and use it as a title:

```typescript
export const productResolver: ResolveFn<Product> = ({ paramMap }) => {
  console.log('resolving product');
  const productService = inject(ProductService);
  const id = paramMap.get('id') ?? 0;
  return productService.getProductById(+id);
};

export const routes: Routes = [
  {
    path: 'product/:id',
    component: ProductComponent,
    title: () => {
      const route = inject(ActivatedRoute);
      return route.data.pipe(
        // this was an attempt to "wait" for the data
        // but it did not work, Observable never emits anything
        // skipWhile((data) => {
        //   return !data['product'];
        // }),
        map((data) => {
          console.log(data?.['product']?.title ?? 'Default');
          return data?.['product']?.title ?? 'Default';
        })
      );
    },
    resolve: { product: productResolver },
  },
];
```

When I access the resolved data via parameters (`title: ({data}) => {}`), I get an empty object. But, and this is important, the `productResolver` works just fine and I am able to read the product data in the component via input binding.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-mfigyo?file=src%2Fmain.ts

### Please provide the exception or error you saw

```true
No exception, just unexpected behavior
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 18.2.0
Node: 18.20.0
Package Manager: npm 10.5.0
OS: win32 x64

Angular: 18.2.0
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1802.0
@angular-devkit/build-angular   18.2.0
@angular-devkit/core            18.2.0
@angular-devkit/schematics      18.2.0
@schematics/angular             18.2.0
rxjs                            7.8.1
typescript                      5.5.4
zone.js                         0.14.10
```

### Anything else?

I am not entirely sure if this is a bug, could be intended behavior (all resolvers including the title resolver run in parallel, resulting in race conditions when trying to resolve the title). However, I would argue that running the title resolver *after* all other resolvers have finished would make more sense here. Also, if this is, in fact, the case, it is strange that even using `skipWhile` to wait does not seem to resolve the issue, it is as if the `data` Observable never emits for the title resolver, which makes me think this might even be a dependency injection issue (the title resolver gets a "wrong" `ActivatedRoute` instance)

**Comment by atscott** [comment]
Resolvers on the same route do run in parallel. I can see an argument for title resolvers running last though you could also have an empty path parent to resolve necessary data, provided there is a fix for this one: #47287. I'm going to close this as superseded by #47287.

> Also, if this is, in fact, the case, it is strange that even using skipWhile to wait does not seem to resolve the issue

The data observable emits once when all data resolves rather than emitting each time a single new item is resolved.

**Comment by Armenvardanyan95** [comment]
@atscott understandable. However, for me, the issue is that the Observable in the `title` resolver never actually emitted. It is possible I did something wrong, but I don't quite see it right now.

I'll take a look at an alternative approach, thank you!
